src/main/java/build/buildfarm/worker/shard/ShardWorkerContext.java

```
@@ -477,30 +476,19 @@ public Duration getMaximumActionTimeout() {
   }
 
   private void insertBlob(Digest digest, ByteString content) throws InterruptedException {
+
     if (digest.getSizeBytes() > 0) {
-      Blob blob = new Blob(content, digest);
-      execFileSystem.getStorage().put(blob);
+      try {
+        writer.get().insertBlob(digest, content);
+      } catch (IOException e) {
+      }
```

Don't swallow this exception